substrate-node/pallets/pallet-smart-contract/src/grid_contract.rs

```diff
@@ -270,9 +270,12 @@ impl<T: Config> Pallet<T> {
         // override values
         contract.contract_type = types::ContractData::NodeContract(node_contract);
 
-        let state = contract.state.clone();
-        Self::update_contract_state(&mut contract, &state)?;
-
+        // Why we call update_contract_state if state can't be changed?
+        // Obviously, we just depened on the update_contract_state for writing the contract to the storage, but we can do it here
+        // let state = contract.state.clone();
+        // Self::update_contract_state(&mut contract, &state)?;
+        Contracts::<T>::insert(&contract.contract_id, contract.clone());
```

Comment:
Good catch
I also don t see the reason why `update_contract_state()` should be called here since `deployment_hash` and `deployment_data` have nothing to do with contract state and what is handled inside `update_contract_state()`.